pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -334,6 +334,9 @@ public void sendComplete(Exception exception) {
                 log.error("[{}] Error producing on remote broker", replicator.replicatorId, exception);
                 // cursor should be rewinded since it was incremented when readMoreEntries
                 replicator.cursor.rewind();
+
+                replicator.cursor.cancelPendingReadRequest();
```

Comment:
Are there any concurrency issues here? Because the pending request is handled in another thread

Reply:
Indeed. Call cancelPendingReadRequest() first and then rewind cursor can reduce it.